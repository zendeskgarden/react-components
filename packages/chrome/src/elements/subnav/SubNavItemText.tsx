/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React, { HTMLAttributes } from 'react';
import PropTypes from 'prop-types';
import { StyledSubNavItemText, IStyledSubNavItemTextProps } from '../../styled';

/**
 * @extends HTMLAttributes<HTMLSpanElement>
 */
export const SubNavItemText = React.forwardRef<
  HTMLElement,
  IStyledSubNavItemTextProps & HTMLAttributes<HTMLSpanElement>
>((props, ref) => <StyledSubNavItemText ref={ref} {...props} />);

SubNavItemText.displayName = 'SubNavItemText';

SubNavItemText.propTypes = {
  isWrapped: PropTypes.bool
};
