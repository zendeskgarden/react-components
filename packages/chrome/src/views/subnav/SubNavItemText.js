/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import PropTypes from 'prop-types';
import styled from 'styled-components';
import classNames from 'classnames';
import { retrieveTheme } from '@zendeskgarden/react-theming';
import ChromeStyles from '@zendeskgarden/css-chrome';

const COMPONENT_ID = 'chrome.subnav_item_text';

/**
 * Accepts all `<span>` props
 */
const SubNavItemText = styled.span.attrs({
  'data-garden-id': COMPONENT_ID,
  'data-garden-version': PACKAGE_VERSION,
  className: props =>
    classNames(ChromeStyles['c-chrome__subnav__item__text'], {
      [ChromeStyles['c-chrome__subnav__item__text--wrap']]: props.wrap
    })
})`
  ${props => retrieveTheme(COMPONENT_ID, props)};
`;

SubNavItemText.propTypes = {
  /** Wrap overflow text instead of truncating long strings with an ellipsis
   (use in conjunction with max-width styling applied to the `SubNav` container) */
  wrap: PropTypes.bool
};

/** @component */
export default SubNavItemText;
