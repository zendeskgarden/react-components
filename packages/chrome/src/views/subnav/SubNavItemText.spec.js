import React from 'react';
import { shallow } from 'enzyme';

import SubNavItemText from './SubNavItemText';

describe('SubNavItemText', () => {
  it('renders default styling', () => {
    const wrapper = shallow(<SubNavItemText />);

    expect(wrapper).toMatchSnapshot();
  });
});
