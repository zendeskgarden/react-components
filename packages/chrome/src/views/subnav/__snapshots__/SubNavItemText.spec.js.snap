// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubNavItemText renders default styling 1`] = `
<span
  className="c-chrome__subnav__item__text "
  data-garden-id="chrome.subnav_item_text"
  data-garden-version="version"
/>
`;
